server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -445,16 +448,14 @@ private void becomeLeader()
               config.getCoordinatorPeriod()
           )
       );
-      if (overlordClient != null) {
```

Comment:
In production code, `OverlordClient` was always non-null. It seems likely that this has been added to handle some test cases in `DruidCoordinatorTest`. But I have fixed that up so that we need not handle the null case anymore.